src/NextGen/DonationForm/Blocks/DonationFormBlock/resources/app/utilities/registerFieldAndBuildProps.tsx

```diff
@@ -0,0 +1,24 @@
+import {Field} from '@givewp/forms/types';
+import {FieldProps} from '@givewp/forms/propTypes';
+import {UseFormReturn} from 'react-hook-form';
+import buildRegisterValidationOptions from './buildRegisterValidationOptions';
+import {getFieldErrorTemplate, getFieldLabelTemplate} from '../templates';
+import getErrorByFieldName from './getErrorByFieldName';
+
+export default function registerFieldAndBuildProps(
+    field: Field,
+    register: UseFormReturn['register'],
```

Comment:
@jonwaldstein @kjohnson This was an interesting find. I wanted the type to be the same type as another type's property. Apparently this is how you do that.